paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/GooglePayButton.kt

```diff
@@ -46,23 +69,99 @@ internal class GooglePayButton @JvmOverloads constructor(
         // Call super so we don't inadvertently effect the primary button as well.
         super.setClickable(true)
         super.setEnabled(true)
-        viewBinding.googlePayPrimaryButton.backgroundTintList = null
-        viewBinding.googlePayPrimaryButton.finishedBackgroundColor = Color.TRANSPARENT
+    }
+
+    fun initialize(
+        cornerRadius: Int,
+        allowCreditCards: Boolean,
+        billingAddressParameters: GooglePayJsonFactory.BillingAddressParameters?
+    ) {
+        initializePrimaryButton()
+        val isDark = context.isSystemDarkTheme()
+        val allowedPaymentMethods = GooglePayJsonFactory(context)
+            .createIsReadyToPayRequest(
+                billingAddressParameters = billingAddressParameters,
+                allowCreditCards = allowCreditCards
+            ).getJSONArray(ALLOWED_PAYMENT_METHODS).toString()
+
+        if (GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(context) != 0) return
```

Comment:
Why do we need this here? The button _should_ only be shown if Google Play is available. If that’s not the case, we should change what we’re testing for in `PaymentSheetLoader`.

Reply:
This change skirts the fact that we are hardcoding `isGooglePayReady` in the PaymentSheetLoader in the PaymentSheetActivityTest, since for some Google Pay tests, we are hardcoding it to true, it will try to render the button, causing some crashes.